src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigNumber.java

```diff
@@ -0,0 +1,103 @@
+ * OneConfig - Next Generation Config Library for Minecraft: Java Edition
+ * Copyright (C) 2021, 2022 Polyfrost.
+ *   <https://polyfrost.cc> <https://github.com/Polyfrost/>
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ *   OneConfig is licensed under the terms of version 3 of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, AND
+ * under the Additional Terms Applicable to OneConfig, as published by Polyfrost,
+ * either version 1.0 of the Additional Terms, or (at your option) any later
+ * version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Lesser General Public
+ * License.  If not, see <https://www.gnu.org/licenses/>. You should
+ * have also received a copy of the Additional Terms Applicable
+ * to OneConfig, as published by Polyfrost. If not, see
+ * <https://polyfrost.cc/legal/oneconfig/additional-terms>
+ */
+
+package cc.polyfrost.oneconfig.gui.elements.config;
+
+import cc.polyfrost.oneconfig.config.annotations.Number;
+import cc.polyfrost.oneconfig.config.elements.BasicOption;
+import cc.polyfrost.oneconfig.gui.elements.text.NumberInputField;
+import cc.polyfrost.oneconfig.renderer.RenderManager;
+import cc.polyfrost.oneconfig.renderer.font.Fonts;
+import cc.polyfrost.oneconfig.utils.InputHandler;
+
+import java.lang.reflect.Field;
+
+public class ConfigNumber extends BasicOption {
+    private final NumberInputField inputField;
+    private boolean isFloat = true;
+
+    public ConfigNumber(Field field, Object parent, String name, String description, String category, String subcategory, float min, float max, int step) {
+        super(field, parent, name, description, category, subcategory, 2);
+        this.inputField = new NumberInputField(84, 32, 0, min, max, step);
+    }
+
+    public static ConfigNumber create(Field field, Object parent) {
+        Number slider = field.getAnnotation(Number.class);
+        return new ConfigNumber(field, parent, slider.name(), slider.description(), slider.category(), slider.subcategory(), slider.min(), slider.max(), slider.step());
+    }
+
+    @Override
+    public void draw(long vg, int x, int y, InputHandler inputHandler) {
+        float value = 0;
+        inputField.disable(!isEnabled());
+        if (!isEnabled()) RenderManager.setAlpha(vg, 0.5f);
+        if (inputField.isToggled() || inputField.arrowsClicked()) {
+            value = inputField.getCurrentValue();
+        }
+        if (inputField.isToggled() || inputField.arrowsClicked()) {
+            setValue(value);
+        }
+
+        if (!inputField.isToggled()) {
+            try {
+                Object object = get();
+                if (object instanceof Integer)
```

Comment:
ngl could be changed to be done in init

Reply:
> ngl could be changed to be done in init

that's just how ConfigSlider is done, someone else can change that in a later PR